src/dvclive/dvc.py

```diff
@@ -98,11 +101,53 @@ def make_dvcyaml(live) -> None:
         dvcyaml["artifacts"] = copy.deepcopy(live._artifacts)
         for artifact in dvcyaml["artifacts"].values():  # type: ignore
             abs_path = os.path.abspath(artifact["path"])
-            abs_dir = os.path.realpath(live.dir)
+            abs_dir = os.path.abspath(dvcyaml_dir)
             relative_path = os.path.relpath(abs_path, abs_dir)
             artifact["path"] = Path(relative_path).as_posix()
 
-    dump_yaml(dvcyaml, live.dvc_file)
+    if not os.path.exists(live.dvc_file):
+        dump_yaml(dvcyaml, live.dvc_file)
+    else:
+        update_dvcyaml(live, dvcyaml)
+
+
+def update_dvcyaml(live, updates):  # noqa: C901
```

Comment:
This contains most of the new logic to update an existing `dvc.yaml` file. The basic logic is:
1. For each section in the old `dvc.yaml`, drop the old dvclive entries
2. Merge the remaining old dvclive entries with the new ones

Reply:
What would be the harm of simplifying the logic and only adding new content if there is no matching "key" in the dvc.yaml? Not trying to drop "outdated" sections, not trying to update props, etc.

We are defining what is the excepted thing here as it is the first time we have done it, so I think it would be okay to start with that and wait for people to ask for more complex update logic.

We would be simplifying code and the number of occasions we would "dynamically" update the `dvc.yaml`

Reply:
> What would be the harm of simplifying the logic and only adding new content if there is no matching "key" in the dvc.yaml?

What do you suggest if there is a matching key? Let's say the artifact labels change. We would at least want to update that artifact, right?

I basically had 2 reasons for doing it this way:
1. It was the closest to mimicking current behavior (we have sort of defined expectations from what we do today in `dvclive/dvc.yaml`).
2. I didn't see how to simplify the logic much more.

Most of the complexity is because:
- metrics, plots, and params are all lists even though they work like dicts (keys must be unique), which makes it hard to merge them. We can't make a set because we could end up with something like `[{"dvclive/plots/custom/roc.json": {"title": "ROC"}}, {"dvclive/plots/custom/roc.json": {"title": "ROC Curve"}}]`, which is invalid for dvc.
- artifacts are easy to merge since they are dicts but they are keyed by name rather than path, so I went through the extra trouble to also make sure we dropped other dvclive paths [here](https://github.com/iterative/dvclive/blob/7e9554a366c59fce7d185193f18b2cc51bbb7a7e/src/dvclive/dvc.py#L120-L122). We could drop this part, but it felt consistent with the rest of the logic and only added 3 lines/1 loop.

Reply:
It makes sense, I couldn't find a simpler way to cover all cases. As discussed offline, let's go with this and see how it works